d2layouts/d2grid/layout.go

```diff
@@ -0,0 +1,578 @@
+package d2grid
+
+import (
+	"context"
+	"math"
+	"sort"
+
+	"oss.terrastruct.com/d2/d2graph"
+	"oss.terrastruct.com/d2/lib/geo"
+	"oss.terrastruct.com/d2/lib/label"
+	"oss.terrastruct.com/util-go/go2"
+)
+
+const (
+	CONTAINER_PADDING = 60
+	HORIZONTAL_PAD    = 40.
+	VERTICAL_PAD      = 40.
```

Comment:
i think eventually we'll want to let users customize gap size too. make issue for that too plz

Reply:
#1146